Not able to build ONNX Runtime Server image

**Describe the bug**
Following https://github.com/microsoft/onnxruntime/tree/master/dockerfiles#onnx-runtime-server, I am not able to build onnxruntime server image. 

I am getting following error:

[ 88%] Building CXX object CMakeFiles/onnxruntime_server_tests.dir/test/unit_tests/prediction_service_impl_test.cc.o
[ 90%] Building CXX object CMakeFiles/onnxruntime_server_tests.dir/test/unit_tests/server_configuration_test.cc.o
[ 92%] Building CXX object CMakeFiles/onnxruntime_server_tests.dir/test/unit_tests/test_main.cc.o
/onnxruntime/server/main.cc:23:44: note: #pragma message: SRV_VERSION=""
 #pragma message(VAR_NAME_VALUE(SRV_VERSION))
                                            ^
/onnxruntime/server/main.cc:29:49: note: #pragma message: LATEST_COMMIT_ID=""
 #pragma message(VAR_NAME_VALUE(LATEST_COMMIT_ID))
                                                 ^
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/memory:63,
                 from /onnxruntime/server/environment.h:6,
                 from /onnxruntime/server/main.cc:4:
/usr/include/c++/5/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = spdlog::sinks::syslog_sink<std::mutex>; _Args = {}; _Tp = spdlog::sinks::syslog_sink<std::mutex>]':
/usr/include/c++/5/bits/alloc_traits.h:530:4:   required from 'static void std::allocator_traits<std::allocator<_Tp1> >::construct(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*, _Args&& ...) [with _Up = spdlog::sinks::syslog_sink<std::mutex>; _Args = {}; _Tp = spdlog::sinks::syslog_sink<std::mutex>; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >]'
/usr/include/c++/5/bits/shared_ptr_base.h:522:39:   required from 'std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {}; _Tp = spdlog::sinks::syslog_sink<std::mutex>; _Alloc = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]'
/usr/include/c++/5/bits/shared_ptr_base.h:617:4:   required from 'std::__shared_count<_Lp>::__shared_count(std::_Sp_make_shared_tag, _Tp*, const _Alloc&, _Args&& ...) [with _Tp = spdlog::sinks::syslog_sink<std::mutex>; _Alloc = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >; _Args = {}; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]'
/usr/include/c++/5/bits/shared_ptr_base.h:1097:35:   required from 'std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_make_shared_tag, const _Alloc&, _Args&& ...) [with _Alloc = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >; _Args = {}; _Tp = spdlog::sinks::syslog_sink<std::mutex>; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2u]'
/usr/include/c++/5/bits/shared_ptr.h:319:64:   required from 'std::shared_ptr<_Tp>::shared_ptr(std::_Sp_make_shared_tag, const _Alloc&, _Args&& ...) [with _Alloc = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >; _Args = {}; _Tp = spdlog::sinks::syslog_sink<std::mutex>]'
/usr/include/c++/5/bits/shared_ptr.h:620:39:   required from 'std::shared_ptr<_Tp1> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = spdlog::sinks::syslog_sink<std::mutex>; _Alloc = std::allocator<spdlog::sinks::syslog_sink<std::mutex> >; _Args = {}]'
/usr/include/c++/5/bits/shared_ptr.h:635:39:   required from 'std::shared_ptr<_Tp1> std::make_shared(_Args&& ...) [with _Tp = spdlog::sinks::syslog_sink<std::mutex>; _Args = {}]'
/onnxruntime/server/main.cc:61:209:   required from here
/usr/include/c++/5/ext/new_allocator.h:120:4: error: no matching function for call to 'spdlog::sinks::syslog_sink<std::mutex>::syslog_sink()'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /onnxruntime/server/main.cc:12:0:
/onnxruntime/server/external/spdlog/include/spdlog/sinks/syslog_sink.h:23:5: note: candidate: spdlog::sinks::syslog_sink<Mutex>::syslog_sink(std::__cxx11::string, int, int, bool) [with Mutex = std::mutex; std::__cxx11::string = std::__cxx11::basic_string<char>]
     syslog_sink(std::string ident, int syslog_option, int syslog_facility, bool enable_formatting)
     ^
/onnxruntime/server/external/spdlog/include/spdlog/sinks/syslog_sink.h:23:5: note:   candidate expects 4 arguments, 0 provided
[ 94%] Building CXX object CMakeFiles/onnxruntime_server_tests.dir/test/unit_tests/test_server_environment.cc.o
make[2]: *** [CMakeFiles/onnxruntime_server.dir/main.cc.o] Error 1
CMakeFiles/onnxruntime_server.dir/build.make:62: recipe for target 'CMakeFiles/onnxruntime_server.dir/main.cc.o' failed
CMakeFiles/Makefile2:254: recipe for target 'CMakeFiles/onnxruntime_server.dir/all' failed
make[1]: *** [CMakeFiles/onnxruntime_server.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 96%] Building CXX object CMakeFiles/onnxruntime_server_tests.dir/test/unit_tests/util_tests.cc.o
[ 98%] Linking CXX executable onnxruntime_server_tests
Running /usr/local/bin/grpc_python_plugin on /onnxruntime/server/protobuf/prediction_service.proto
[ 98%] Built target onnxruntime_server_tests
make: *** [all] Error 2
Makefile:83: recipe for target 'all' failed
The command '/bin/sh -c mkdir -p /onnxruntime/build && cd /onnxruntime/build && cmake -DCMAKE_BUILD_TYPE=Release /onnxruntime/server     && make -j$(getconf _NPROCESSORS_ONLN)' returned a non-zero code: 2

**Urgency**
None

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04

**To Reproduce**
Running docker build command as per documentation:
docker build -t {docker_image_name} -f Dockerfile.server .

**Expected behavior**
Successful building of onnxruntime server image.

--- a/.gitmodules
+++ b/.gitmodules
@@ -52,6 +52,7 @@
 [submodule "server/external/spdlog"]
 	path = server/external/spdlog
 	url = https://github.com/gabime/spdlog.git
+	branch = master
 [submodule "cmake/external/FeaturizersLibrary"]
 	path = cmake/external/FeaturizersLibrary
 	url = https://github.com/microsoft/FeaturizersLibrary.git

